File: hw5/Solution/m-files.d/ex2.m
clc; clear all; close all;

f = fscanf(fopen('f.dat','r'), '%f');
N = length(f);
x0 = 101.15520937701599;

lSet = (2:16)';
ft = zeros(N, numel(lSet));
err = zeros(size(lSet));

for i = 1:numel(lSet)

    l = lSet(i);
    idx = ((1:2^l)-1)*2^(16-l)+1;
    
    [fhat, Ahat, Bhat, fig(i), ft(:,i)] = FT(f(idx));

    err(i) = max(abs(ft(:,i) - f));

end % for-loop

DataSet = 2.^lSet;
T = table(DataSet, err);
disp(T);

figure(1);
plot(2*x0*(0:(N-1))'/N, ft);
grid on;
set(legend('$T_2(t)$','$T_4(t)$','$T_6(t)$',...
             '$T_8(t)$','$T_{10}(t)$','$T_{12}(t)$','$T_{14}(t)$','$T_{16}(t)$'),...
    'Interpreter','latex');

File: hw5/Solution/py-files.d/ex2.py
import numpy, matplotlib.pyplot
import FunHW5

f = numpy.loadtxt('f.dat')
N = f.size
x0 = 101.15520937701599

lSet = numpy.arange(2,16,2)
ft = numpy.zeros((N, lSet.size))
err = numpy.zeros(lSet.shape)

for i in range(lSet.size):

    l = lSet[i]
    idx = numpy.arange(2**l)*2**(16-l)

    fhat, Ahat, Bhat, figi, ft[:,i] = FunHW5.FT(f[idx])

    err[i] = max(abs(ft[:,i] - f))
    
matplotlib.pyplot.show()
print('\n')
print('error:\n' + str(err))
